packages/core/src/libraries/sign-in-experience/index.ts

```diff
@@ -164,12 +190,14 @@ export const createSignInExperienceLibrary = (
       isDevelopmentTenant,
       organizationOverride,
       appSignInExperience,
+      captchaConfig,
     ] = await Promise.all([
       findDefaultSignInExperience(),
       getLogtoConnectors(),
       getIsDevelopmentTenant(),
       getOrganizationOverride(organizationId),
       findApplicationSignInExperience(appId),
+      findCaptchaPublicConfig(),
```

Comment:
By default, our database connection pool allows a maximum of 10 active connections per tenant. Excessive asynchronous queries may lead to performance issues.  Just like what we have encountered in the cloud repo, when calculating tenant usage, we have to use one single connection to run the queries instead of running the multi query method in async.  This code logic here is very risky.

Reply:
Created LOG-11055 to track, will find a way to handle this before launch.